tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -540,8 +537,15 @@ void EnqueueProgramCommand::assemble_runtime_args_commands() {
 
     this->cached_program_command_sequences[program.id].runtime_args_command_sequences.reserve(command_count);
     // Unique Runtime Args (Unicast)
-    for (CoreType core_type : core_types) {
-        for (auto& kg : program.get_kernel_groups(core_type)) {
+    for (uint32_t index = 0; index < hal.get_programmable_core_type_count(); index++) {
+        if (hal.get_programmable_core_type(index) == HalProgrammableCoreType::IDLE_ETH) {
+            // Fast dispatch not supported on IDLE_ETH yet
+            // TODO: can't just loop here as code below confuses ACTIVE/IDLE
```

Comment:
I assume replacing CoreType with ProgrammableCoreType is in follow up PR

Reply:
I don't follow

Reply:
oh my comment was for a below line, we pull out CoreType from the HAL and then have logic based on that, but the logic here can be updated to be based on ProgrammableCoreType

Reply:
(hmm I don't know why I can't reply to your comment below)
AddKernel is an internal detail api. I'm going to change it and if we change again no big deal.  As far as I can tell, only tt_metal.cpp calls it